### PR TITLE
feat: use Postgres for clustering strategy

### PR DESCRIPTION
**Pull request by Ziinc**
This PR uses postgres for clustering, which will base the notify channel name on the magic cookie.

### CODE REVIEW

File: lib/logflare/cluster/postgres_strategy.ex [hunk 1]
```diff
@@ -0,0 +1,117 @@
+defmodule Logflare.Cluster.PostgresStrategy do
```
Comment (hauleth) [inline lib/logflare/cluster/postgres_strategy.ex:1]: Maybe we could extract that and publish as a separate library? I think that quite a few people may find it useful.

**Review by hauleth — APPROVED** [review]

File: lib/logflare/cluster/postgres_strategy.ex (hunk 1 above)
Comment (chasers) [reply lib/logflare/cluster/postgres_strategy.ex:1]: We definitely should.  Realtime and Supavisor are using it.

File: lib/logflare/cluster/postgres_strategy.ex [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+
+  When a node comes online, it begins to broadcast its name in a "heartbeat" message to the channel. All other nodes that receive this message attempt to connect to it.
+
+  This strategy does not check connectivity between nodes and does not disconnect them
+
+  ## Options
+
+  * `heartbeat_interval` - The interval at which to send heartbeat messages in milliseconds (optional; default: 5_000)
+
+  The magic cookie is used as the channel name.
```
Comment (chasers) [inline lib/logflare/cluster/postgres_strategy.ex:15]: Did we check constraints around this from Postgres?

Comment (chasers) [reply lib/logflare/cluster/postgres_strategy.ex:15]: In Supavisor and Realtime this is passed in with the config.

If we want to use the magic cookie we should probably move this and do it there.

Comment (Ziinc) [reply lib/logflare/cluster/postgres_strategy.ex:15]: from my manual testing, pg doesn't accept a channel name with `:` prefix, hence why i converted it to use `Atom.to_string/1` to get the stringified atom. `Node.get_cookie/0` seems to return the cookie as an atom.
I can make this more foolproof though with some alphanumeric conversions

Comment (chasers) [reply lib/logflare/cluster/postgres_strategy.ex:15]: Roger.  Maybe default behavior (for the eventual library) is to use the cleaned magic cookie and override (or additionally) subscribe to the channel(s) passed in the config.

Comment (chasers) [reply lib/logflare/cluster/postgres_strategy.ex:15]: And yeah just double check because the cookie on dev could not be as long or complex as prod.

**Review by chasers — APPROVED** [review]

File: lib/logflare/cluster/postgres_strategy.ex (hunk 1 above)
Comment (Ziinc) [reply lib/logflare/cluster/postgres_strategy.ex:1]: task [here](https://www.notion.so/supabase/Logflare-Clustering-Extract-out-Postgres-Clustering-Strategy-to-its-own-separate-library-42d88bf841ad4338a19daac2aed11e25?pvs=4), backlogged

**Pull request merged**